waku-bindings/src/decrypt.rs

```diff
@@ -16,23 +16,33 @@ pub fn waku_decode_symmetric(
     symmetric_key: &Key<Aes256Gcm>,
 ) -> Result<DecodedPayload> {
     let symk = hex::encode(symmetric_key.as_slice());
-    let result = unsafe {
-        CStr::from_ptr(waku_sys::waku_decode_symmetric(
-            CString::new(
-                serde_json::to_string(&message)
-                    .expect("WakuMessages should always be able to success serializing"),
-            )
-            .expect("CString should build properly from the serialized waku message")
-            .into_raw(),
-            CString::new(symk)
-                .expect("CString should build properly from hex encoded symmetric key")
-                .into_raw(),
-        ))
-    }
-    .to_str()
-    .expect("Response should always succeed to load to a &str");
+
+    let message_ptr = CString::new(
+        serde_json::to_string(&message)
+            .expect("WakuMessages should always be able to success serializing"),
+    )
+    .expect("CString should build properly from the serialized waku message")
+    .into_raw();
```

Comment:
`CString::new().as_cstr().as_ptr()` is maybe better as it does not transfer ownership nor require to call `from_raw` at the end to drop the string, provided waku does not change those strings

Reply:
But the problem with that is I get a `* const T` when it requires a `*mut T`.  So then I need to recast the whole thing, `.as_cstr().as_ptr() as *mut _`. So, we do not transfer ownership but make the raw conversion anyway.
Both are ugly 💩 . It is true that we can skip the manual drop that way.

Reply:
if the C function is not modifying the string the arg should be a const ptr, and we need to fix it

Reply:
Yes, I understand. This is autogenerated `cbindgen` from autogenerated code go `"C"` stuff. Lets open an issue on this as it may take more than one effort to change.